src/twisted/mail/test/test_mail.py

```diff
@@ -50,11 +55,6 @@
 from twisted.python import failure, log
 from twisted.python.filepath import FilePath
 from twisted.python.runtime import platformType
-from twisted.test.proto_helpers import (
-    LineSendingProtocol,
-    MemoryReactorClock,
-    StringTransport,
-)
```

Comment:
This looks like it is correct but this test module runs nowhere because no one ever ported it to Python 3.  Before making these changes, it would make sense to get the tests in this module to pass on Python 3.

Reply:
Against trunk, I see that only 33/103 tests are skipped on python 3.10.
`(twstd310) eevel@weezel3:~/twisted$ trial twisted.mail.test.test_mail
twisted.mail.test.test_mail
  AbstractMaildirDomainTests
    test_interface ...                                                     [OK]
  AddressAliasTests
    test_createMessageReceiver ...                                         [OK]
    test_resolve ...                                                       [OK]
    test_resolveWithoutAliasmap ...                                        [OK]
    test_str ...                                                           [OK]
  AliasTests
    testFileAlias ...                                                 [SKIPPED]
    testFileLoader ...                                                [SKIPPED]
    testHandle ...                                                         [OK]
    testMultiWrapper ...                                                   [OK]
  BounceTests
    testAddUser ...                                                        [OK]
    testExists ...                                                         [OK]
    testRelay ...                                                          [OK]
  BounceWithSMTPServerTests
    test_rejected ...                                                      [OK]
  DirectoryQueueTests
    testDone ...                                                      [SKIPPED]
    testEnvelope ...                                                       [OK]
    testRelaying ...                                                  [SKIPPED]
    testWaiting ...                                                   [SKIPPED]
  DomainWithDefaultsTests
    testMethods ...                                                   [SKIPPED]
    test_has_keyDeprecation ...                                            [OK]
    test_repr ...                                                          [OK]
    test_str ...                                                           [OK]
  FileMessageTests
    testContents ...                                                       [OK]
    testFinalName ...                                                      [OK]
    testInterrupted ...                                                    [OK]
  LiveFireExerciseTests
    testLocalDelivery ...                                             [SKIPPED]
    testRelayDelivery ...                                             [SKIPPED]
  MXTests
    testManyRecords ...                                               [SKIPPED]
    testSimpleFailure ...                                                  [OK]
    testSimpleFailureWithFallback ...                                      [OK]
    testSimpleSuccess ...                                             [SKIPPED]
    test_badExchangeExcluded ...                                           [OK]
    test_badExchangeExpires ...                                            [OK]
    test_cnameChain ...                                                    [OK]
    test_cnameLoopWithGlueRecords ...                                      [OK]
    test_cnameWithGlueRecords ...                                          [OK]
    test_cnameWithoutGlueRecords ...                                       [OK]
    test_defaultClock ...                                                  [OK]
    test_failureWithSuccessfulFallback ...                                 [OK]
    test_fallbackForAllBadExchanges ...                                    [OK]
    test_goodExchangeUsed ...                                              [OK]
    test_mailExchangePreference ...                                        [OK]
    test_successWithoutResults ...                                         [OK]
  MailServiceTests
    testFactories ...                                                      [OK]
    testPortals ...                                                        [OK]
  MaildirAppendFileTests
    test_append ...                                                   [SKIPPED]
  MaildirAppendStringTests
    test_append ...                                                   [SKIPPED]
  MaildirDirdbmDomainTests
    test_addUser ...                                                  [SKIPPED]
    test_credentials ...                                                   [OK]
    test_requestAvatar ...                                            [SKIPPED]
    test_requestAvatarId ...                                          [SKIPPED]
    test_userDirectory ...                                            [SKIPPED]
  MaildirMessageTests
    test_contents ...                                                      [OK]
    test_finalName ...                                                     [OK]
    test_interrupted ...                                                   [OK]
  MaildirTests
    testInitializer ...                                                    [OK]
    test_mailbox ...                                                  [SKIPPED]
    test_nameGenerator ...                                                 [OK]
  ManagedRelayerTests
    testConnectionLost ...                                                 [OK]
    testFailedSentMail ...                                                 [OK]
    testSuccessfulSentMail ...                                             [OK]
  ProcessAliasTests
    test_aliasResolution ...                                          [SKIPPED]
    test_cyclicAlias ...                                              [SKIPPED]
    test_earlyProcessTermination ...                                       [OK]
    test_errorProcessTermination ...                                       [OK]
    test_processAlias ...                                             [SKIPPED]
    test_processAliasTimeout ...                                           [OK]
    test_signalProcessTermination ...                                      [OK]
  RelayTests
    testExists ...                                                    [SKIPPED]
  RelayerTests
    testMailData ...                                                       [OK]
    testMailFrom ...                                                       [OK]
    testMailTo ...                                                         [OK]
  ServiceDomainTests
    testAddAliasableDomain ...                                             [OK]
    testReceivedHeader ...                                            [SKIPPED]
    testValidateFrom ...                                                   [OK]
    testValidateTo ...                                                [SKIPPED]
    testValidateToBadDomain ...                                            [OK]
    testValidateToBadUsername ...                                          [OK]
  StringListMailboxTests
    test_deleteMessage ...                                                 [OK]
    test_getMessage ...                                               [SKIPPED]
    test_getUidl ...                                                  [SKIPPED]
    test_listAllMessages ...                                               [OK]
    test_listOneMessage ...                                                [OK]
    test_sync ...                                                          [OK]
    test_undeleteMessages ...                                              [OK]
  VirtualPOP3Tests
    testAuthenticateAPOP ...                                          [SKIPPED]
    testAuthenticateBadPasswordPASS ...                               [SKIPPED]
    testAuthenticateBadUserPASS ...                                   [SKIPPED]
    testAuthenticateIncorrectResponseAPOP ...                         [SKIPPED]
    testAuthenticateIncorrectUserAPOP ...                             [SKIPPED]
    testAuthenticatePASS ...                                          [SKIPPED]
  _AttemptManagerTests
    test_initNoisy ...                                                     [OK]
    test_initNoisyDefault ...                                              [OK]
    test_initQuiet ...                                                     [OK]
    test_initReactor ...                                                   [OK]
    test_initReactorDefault ...                                            [OK]
    test_notifyDoneNoisy ...                                               [OK]
    test_notifyDoneQuiet ...                                               [OK]
    test_notifyFailureNoisy ...                                       [SKIPPED]
    test_notifyFailureQuiet ...                                       [SKIPPED]
    test_notifyNoConnectionNoisy ...                                       [OK]
    test_notifyNoConnectionQuiet ...                                       [OK]
    test_notifySuccessNoisy ...                                            [OK]
    test_notifySuccessQuiet ...                                            [OK]

===============================================================================
[SKIPPED]
not ported to Python 3

twisted.mail.test.test_mail.AliasTests.testFileAlias
twisted.mail.test.test_mail.AliasTests.testFileLoader
twisted.mail.test.test_mail.DirectoryQueueTests.testDone
twisted.mail.test.test_mail.DirectoryQueueTests.testRelaying
twisted.mail.test.test_mail.DirectoryQueueTests.testWaiting
twisted.mail.test.test_mail.DomainWithDefaultsTests.testMethods
twisted.mail.test.test_mail.LiveFireExerciseTests.testLocalDelivery
twisted.mail.test.test_mail.LiveFireExerciseTests.testRelayDelivery
twisted.mail.test.test_mail.MXTests.testManyRecords
twisted.mail.test.test_mail.MXTests.testSimpleSuccess
twisted.mail.test.test_mail.MaildirAppendFileTests.test_append
twisted.mail.test.test_mail.MaildirAppendStringTests.test_append
twisted.mail.test.test_mail.MaildirDirdbmDomainTests.test_addUser
twisted.mail.test.test_mail.MaildirDirdbmDomainTests.test_requestAvatar
twisted.mail.test.test_mail.MaildirDirdbmDomainTests.test_requestAvatarId
twisted.mail.test.test_mail.MaildirDirdbmDomainTests.test_userDirectory
twisted.mail.test.test_mail.MaildirTests.test_mailbox
twisted.mail.test.test_mail.ProcessAliasTests.test_aliasResolution
twisted.mail.test.test_mail.ProcessAliasTests.test_cyclicAlias
twisted.mail.test.test_mail.ProcessAliasTests.test_processAlias
twisted.mail.test.test_mail.RelayTests.testExists
twisted.mail.test.test_mail.ServiceDomainTests.testReceivedHeader
twisted.mail.test.test_mail.ServiceDomainTests.testValidateTo
twisted.mail.test.test_mail.StringListMailboxTests.test_getMessage
twisted.mail.test.test_mail.StringListMailboxTests.test_getUidl
twisted.mail.test.test_mail.VirtualPOP3Tests.testAuthenticateAPOP
twisted.mail.test.test_mail.VirtualPOP3Tests.testAuthenticateBadPasswordPASS
twisted.mail.test.test_mail.VirtualPOP3Tests.testAuthenticateBadUserPASS
twisted.mail.test.test_mail.VirtualPOP3Tests.testAuthenticateIncorrectResponseAPOP
twisted.mail.test.test_mail.VirtualPOP3Tests.testAuthenticateIncorrectUserAPOP
twisted.mail.test.test_mail.VirtualPOP3Tests.testAuthenticatePASS
twisted.mail.test.test_mail._AttemptManagerTests.test_notifyFailureNoisy
twisted.mail.test.test_mail._AttemptManagerTests.test_notifyFailureQuiet
-------------------------------------------------------------------------------
Ran 103 tests in 0.122s

PASSED (skips=33, successes=70)
`

Reply:
I see that the mail tests are executed, so I think is ok to have this change.

https://github.com/twisted/twisted/actions/runs/3319434733/jobs/5484586919#step:10:6367

@exarkun is this still a blocker for merging this PR ?

Thanks

Reply:
It seems like a blocker to me.  It's hard to care _that_ much because it seems pretty clear that there are no real users of Twisted Mail at this point.  I am not going to block this PR on this point - but I am also not going to approve it.